src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -721,6 +732,15 @@ private bool IsAuthenticationError(Exception ex)
             return false;
         }
 
+        private void UpdateCacheExpirationForWatchers(bool success)

```

Comment:
I don't think we do this in unsuccessful case anymore. In that case we may not need the boolean.

Reply:
Same for `UpdateCacheExpirationTime`

Reply:
This method is really just holding the control flow looping over the watchers, and it's only called in one place now. May be a good candidate to inline.

Reply:
Thanks for pointing out!
Updated!